gate-web/src/main/groovy/com/netflix/spinnaker/gate/config/PostConnectionConfiguringJedisConnectionFactory.java

```
@@ -63,6 +71,31 @@ public PostConnectionConfiguringJedisConnectionFactory(
 
     if (redisUri.getScheme().equals("rediss")) {
       setUseSsl(true);
+      String jksFilePath = "/opsmx/conf/redis-truststore.jks";
+      String alias = "redis-truststore"; // An alias to identify the certificate in the keystore
+      char[] password = this.password.toCharArray(); // Keystore password
+
+      try {
+        CertificateFactory certificateFactory = CertificateFactory.getInstance("X.509");
+        FileInputStream certInputStream = new FileInputStream(certFilePath);
+        Certificate certificate = certificateFactory.generateCertificate(certInputStream);
+        certInputStream.close();
+
+        KeyStore keyStore = KeyStore.getInstance("JKS");
+        keyStore.load(null, password);
+        keyStore.setCertificateEntry(alias, certificate);
+
+        FileOutputStream jksOutputStream = new FileOutputStream(jksFilePath);
+        keyStore.store(jksOutputStream, password);
+        jksOutputStream.close();
+
+        System.out.println("Certificate has been added to the KeyStore successfully.");
+      } catch (Exception e) {
+        e.printStackTrace();
```

There is no point in proceeding with service initialization when an exception is caught, so please throw the exception appropriately.